### ISSUE TITLE
Authentication Token not persisted

### ISSUE THREAD

**Issue by vHanda** [watercooler-labs/toggl-cli#81]
For the past couple of months, after I restart my machine, I need to always re-authenticate `toggl` via the `toggl auth <key>` command.

I'm not sure if this means something in my hard-drive is getting corrupted or if toggl isn't storing it in disk any more and is just storing it in some temporary directory.

Would there be an easy way to know where it is storing this authentication information? I'm on linux (Arch).

**Comment by vHanda** [comment]
Checking the code, I do see that it's using [keyring](https://docs.rs/keyring/latest/keyring/), which uses [keyutils](https://man7.org/linux/man-pages/man7/keyutils.7.html). Not sure what I'll need to do to make it persistent across boots.

**Comment by shantanuraj** [comment]
Hey @vHanda that's unfortunate, I'll try and replicate on a vm, meanwhile could you let us know the desktop environment you're using it with or if you're using it headless.

**Comment by shantanuraj** [comment]
Okay looked into the `keyring` package this is from the docs 

> # Persistence
> The key management facility provided by the kernel is completely in-memory and will not persist
> across reboots. Consider the keyring a secure cache and plan for your application to handle
> cases where the entry is no-longer available in-memory.
> In general you should prepare for `Entry::get_password` to fail and have a fallback to re-load
> the credential into memory.
> Potential options to re-load the credential into memory are:
> - Re-prompt the user (most common/effective for CLI applications)
> - Create a PAM module or use `pam_exec` to load a credential securely when the user logs in.
> - If you're running as a systemd service you can use `systemd-ask-password` to prompt the user
>   when your service starts.

https://github.com/hwchen/keyring-rs/blob/779dfe0dc3ebaf5b0bba04864ddc9c32d4a7c3d0/src/keyutils.rs#L19-L33

so it seems like we might have to rework this for linux I'll try and address it this weekend.

**Comment by vHanda** [comment]
Thank you. That's really kind of you. But really, I'm happy to use the software as in - It clearly isn't your use-case.

Plus, I got to learn about something - linux keyring.

Thank you for providing this!

**Comment by wd60622** [comment]
I'm having the same issue on WSL. Could it be possible to pull from environment variable as quick alternative? i.e. `TOGGL_API_TOKEN`

**Comment by 0atman** [comment]
I'm also having this issue on linux, an env var (as suggested by @wd60622) would be a super flexible way to not only fix this issue but sidestep quite a few future ones.
Thank you so much @shantanuraj, it's still a really great tool even with having to re-auth every reboot! :grin:

**Comment by shantanuraj** [comment]
Hey apologies for not getting to this sooner, I'll implement a fallback with a `TOGGL_API_TOKEN` environment variable, while we look into a proper fix for keyring on linux.

**Comment by shantanuraj** [comment]
I've published version 0.4.8, it should allow to use your api tokens via the `TOGGL_API_TOKEN` environment variable.

```bash
TOGGL_API_TOKEN=xxxxx toggl list
```

Thanks for your patience!

**Comment by 0atman** [comment]
It works! Thank you so much :grin: 

```sh
❯ toggl
Please set your API token first by calling toggl auth <API_TOKEN>. Or export it as TOGGL_API_TOKEN.
You can find your API token at https://track.toggl.com/profile

❯ toggl auth ******

❯ TOGGL_API_TOKEN=****** toggl
No time entry is running at the moment
```
